ofrak_core/test_ofrak/components/test_filesystem_component.py

```diff
@@ -28,83 +30,90 @@
+    await filesystem_root.initialize_from_disk(filesystem_root_directory)
+    yield filesystem_root
 
 
 class TestFilesystemRoot:
     """
     Test FilesystemRoot methods.
     """
 
-    async def test_initialize_from_disk(self, ofrak_context: OFRAKContext):
+    async def test_initialize_from_disk(
+        self, ofrak_context: OFRAKContext, filesystem_root_directory
+    ):
         """
         Test that FilesystemRoot.initialize_from_disk modifies a resources tree summary.
         """
-        with FilesystemRootDirectory() as temp_dir:
-            resource = await ofrak_context.create_root_resource(
-                name=temp_dir, data=b"", tags=[FilesystemRoot]
-            )
-            original_tree = await resource.summarize_tree()
-            filesystem_root = await resource.view_as(FilesystemRoot)
-            await filesystem_root.initialize_from_disk(temp_dir)
-            initialized_tree = await resource.summarize_tree()
-            assert original_tree != initialized_tree
+        resource = await ofrak_context.create_root_resource(
+            name=filesystem_root_directory, data=b"", tags=[FilesystemRoot]
+        )
+        original_tree = await resource.summarize_tree()
+        filesystem_root = await resource.view_as(FilesystemRoot)
+        await filesystem_root.initialize_from_disk(filesystem_root_directory)
+        initialized_tree = await resource.summarize_tree()
+        assert original_tree != initialized_tree
 
-    async def test_flush_to_disk(self, ofrak_context: OFRAKContext):
+    async def test_flush_to_disk(self, ofrak_context: OFRAKContext, filesystem_root_directory):
         """
         Test that FilesystemRoot.flush_to_disk correctly flushes the filesystem resources.
         """
-        with FilesystemRootDirectory() as temp_dir:
-            resource = await ofrak_context.create_root_resource(
-                name=temp_dir, data=b"", tags=[FilesystemRoot]
-            )
-            filesystem_root = await resource.view_as(FilesystemRoot)
-            await filesystem_root.initialize_from_disk(temp_dir)
+        resource = await ofrak_context.create_root_resource(
+            name=filesystem_root_directory, data=b"", tags=[FilesystemRoot]
+        )
+        filesystem_root = await resource.view_as(FilesystemRoot)
+        await filesystem_root.initialize_from_disk(filesystem_root_directory)
 
-            with tempfile.TemporaryDirectory() as flush_dir:
-                await filesystem_root.flush_to_disk(flush_dir)
+        with tempfile.TemporaryDirectory() as flush_dir:
+            await filesystem_root.flush_to_disk(flush_dir)
 
-                diff_directories(temp_dir, flush_dir, extra_diff_flags="")
+            if sys.platform != "win32":
+                diff_directories(filesystem_root_directory, flush_dir, extra_diff_flags="")
+            else:
+                warnings.warn(
+                    "Directories not compared on Windows. TODO: implement a basic comparison"
```

Comment:
Probably makes sense to open an issue for this if you're not planning to include it in this PR.

Reply:
opened #531